crates/trie/sparse/src/trie.rs

```diff
@@ -206,6 +206,82 @@ impl<P> fmt::Debug for RevealedSparseTrie<P> {
     }
 }
 
+/// Turns a [`Nibbles`] into a [`String`] by converting each nibble to its hex representation.
+fn encode_nibbles(nibbles: &Nibbles) -> String {
+    let mut output = String::with_capacity(nibbles.len().div_ceil(2));
+    for nibble in nibbles.as_slice() {
+        let hex = format!("{nibble:01x}");
+        output.push_str(&hex);
+    }
+    output
+}
```

Comment:
hmm won't this be confusing because `10` can be interpreted as both `0100` and `a`

maybe just do `hex::encode(nibbles.pack())`?

Reply:
doing `hex::encode(nibbles.pack())` incorrectly encodes paths with odd length (we get an extra `0` at the end). I'm also expecting paths to be interpreted as hex, so I'm not sure why `10` should be interpreted as `a`?

Reply:
just pop the char at the end if len % 2 != 0, idk what you mean with the second sentence